src/util.ts

```diff
@@ -120,7 +120,34 @@ function createClientCredsWithCustomCert(
   );
 }
 
+function promisifyStream<P1, P2, P3>(fn: (req: P1) => grpc.ClientReadableStream<P2>, bind: P3): (req: P1) => Promise<P2[]> {
```

Comment:
Can we bind the generics here to the protocol buffer base interface(s)?

Reply:
@josephschorr I may be misunderstanding this comment - the `ZedPromiseClientInterface` takes care of inferring the generics from the base interfaces based on the matching declarations (for both callbacks and streams). The usage of these methods will have the correct typings from the base methods.

I chose to go this route over manually recreating the interfaces as it should allow for automatic support of new client methods in the future, or until grpc-js supports promises directly: https://github.com/grpc/grpc-node/issues/54

Reply:
I was just curious if there was a way to further restrict the types of P1-P3 so that they only work with our specific response types; its not critical if the answer is "not easily"

Reply:
Thanks for clarifying. Honestly, I don't think so because the generated types aren't inheriting any common base interfaces, so it would be moreso restricting P1-P3 to very specific cases manually, which doesn't buy much. This utility method is not exposed publicly, so the final methods (i.e. the generated promise `lookupResources`) will be fully-typed to that of the original base types via the inferred interface https://github.com/authzed/authzed-node/pull/47/files/742028b528add5c02040ce913e0ded43ab5877f9#diff-ac70eb13bcaa386d4139968851e423915ee41c96e70d29b1d78c7659e1aa0bd2R21

Reply:
Just as a sanity check, do you see all the method signatures inferred correctly? I'm only seeing v1 methods that take a single request object.

Reply:
@samkim you're right - I missed that as part of the type inference. I have a commit I'll push up in the next few hours that should set them correctly. Thanks!

Reply:
Ah ok, can you include a call using one of the other signatures in a test eg https://github.com/authzed/authzed-node/pull/47/files#diff-20dc1a7112d38b8a2aa3c687aa523851d6f5b7c62acc2e08274d2be289ec86d4R73

After that, I'll merge and publish a release. Thanks!

Reply:
@samkim see the latest commit [3093dbf](https://github.com/authzed/authzed-node/pull/47/commits/3093dbf875cca82f97033e58a1a200c7c56031c8)

I restructured the typescript shape to be more specific to the grpc types and allow the full signature to be passed. You can see the full signatures in use in the `v1-promises.test.ts` test.